### PR TITLE
Add .editorconfig + travis hook

### PR DESCRIPTION
**Pull request by rm-hull**
[+ fixed up files that violated the rules]

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+[*]
+charset = utf-8
+trim_trailing_whitespace = true
+end_of_line = lf
+insert_final_newline = true
+
+[*.{clj,cljs,cljc}]
+indent_style = space
+indent_size = 2
```
Comment (lvh) [inline .editorconfig]: There is no newline here; which should break CI because this file is non-compliant

**Comment by lvh** [comment]
Looks good modulo comment

**Comment by arrdem** [comment]
😄  editorconfigs

**Comment by rm-hull** [comment]
Ok, so setting indent_size = 2 is probably not what we want for .clj sources.
And expanded the travis config globbing to `.* **` to pick up hidden files and descend into subdirectories properly

**Comment by lvh** [comment]
In my existing clj editorconfig, I set `indent_size = default` (because I set `indent_size` for `*`).

**Comment by rm-hull** [comment]
I set `root=true` - presumably then if you have an _~/.editorconfig_, it won't take any of your settings?

**Comment by lvh** [comment]
Yes. (That's mostly good.)

**Comment by arrdem** [comment]
cljfmt has a dry-run mode, would that be a better whitespace linter?

**Comment by lvh** [comment]
We should _also_ run cljfmt, yes; but I think that's a separate issue.

**Pull request merged**